Documentation/ApiOverview/Environment/ApplicationContext.rst

```diff
@@ -0,0 +1,383 @@
+    look at the following examples below.
+
+..  _set-application-context-nginx:
+
+Nginx
+-----
+
+..  code-block::
+    :caption: /etc/nginx/nginx.conf
+
+    fastcgi_param TYPO3_CONTEXT Development;
+
+..  _set-application-context-env:
+
+.env (Composer only)
+--------------------
+
+It is possible to import `.env` files into the root directory of your project.
+All contained values are then made available as environment variables. The
+basis for this is the Symfony `.env loader` :composer:`symfony/dotenv`. However,
+this package requires a few method calls for initialisation. You can either
+build this yourself or use the
+`HelHum .env connector` :composer:`helhum/dotenv-connector`. This will
+initialise the Symfony package for you.
+
+Installation
+~~~~~~~~~~~~
+
+..  code-block::
+
+    composer req helhum/dotenv-connector
+
+.env
+~~~~
+
+Please make sure not to insert any spaces before and after the `=`
+
+..  code-block:: bash
+    :caption: `[web root]/.env`
+
+    TYPO3_CONTEXT=Development/Dev1
+
+..  _set-application-context-autoloader:
+
+AutoLoader (Composer only)
+--------------------------
+
+If your TYPO3 was set up using Composer, you can misuse the
+`Composer files <https://getcomposer.org/doc/04-schema.md#files>`__
+property to load a specific file with each request before all other files.
+
+..  tabs::
+
+    ..  group-tab:: composer.json
+
+        ..  code-block:: json
+
+            {
+                'autoload': {
+                    'files': ['Source/Scripts/ApplicationContext.php']
+                }
+            }
+
+    ..  group-tab:: ApplicationContext.php
+
+        ..  code-block:: php
+
+            <?php
+            putenv('TYPO3_CONTEXT=Development');
+
+..  _set-application-context-php-ini:
+
+php.ini
+-------
+
+In `php.ini` there is the option of always loading a specific file first for
+each request. The property is
+`auto_prepend_file <https://www.php.net/manual/en/ini.core.php#ini.auto-prepend-file>`.
+Enter the absolute path to a php file with the following content in your
+hosting package.
+
+..  code-block:: php
+
+    <?php
+    putenv('TYPO3_CONTEXT=Development');
+
+..  attention::
+    Some hosters do not offer the option of setting this value in their
+    configuration menus. If the hoster uses Plesk, you can try to make this
+    setting in a `user.ini` in the document root. Alternatively, try
+    `.user.ini`, `php.ini` and also `.php.ini`. If in doubt, ask the hoster
+    what alternative options are available.
+
+..  _set-application-context-index-php:
+
+index.php
+---------
+
+Please use this version as a last resort if none of the previous versions have
+worked or if your hoster has restricted you enormously. This solution only
+works for the frontend. This means that the `ApplicationContext` displayed in
+the TYPO3 backend may differ from the `ApplicationContext` actually used in
+the frontend.
+
+Create your own `index.php` in the document root directory and then load the
```

Comment:
We have worked with that solution since ages. There is no need to update anything as the real `index.php` will be required: `require_once('typo3_src/index.php')`. The only downside is explained.